vlm/src/main/scala/geotrellis/contrib/vlm/spark/RasterSourceRDD.scala

```diff
@@ -40,78 +41,82 @@ object RasterSourceRDD {
   def read(
     readingSources: Seq[ReadingSource],
     layout: LayoutDefinition,
-    partitionBytes: Long
-  )(implicit sc: SparkContext): MultibandTileLayerRDD[SpatialKey] = {
-    val cellTypes = readingSources.map { _.source.cellType }.toSet
-    require(cellTypes.size == 1, s"All RasterSources must have the same CellType, but multiple ones were found: $cellTypes")
-
-    val projections = readingSources.map { _.source.crs }.toSet
-    require(
-      projections.size == 1,
-      s"All RasterSources must be in the same projection, but multiple ones were found: $projections"
-    )
+    keyTransform: (RasterSource, SpatialKey) => SpaceTimeKey
+  )(implicit sc: SparkContext): MultibandTileLayerRDD[SpaceTimeKey] =
+    read(readingSources, layout, keyTransform, DEFAULT_PARTITION_BYTES)
 
-    val cellType = cellTypes.head
-    val crs = projections.head
+  def read(
```

Comment:
Added lots of fancy overloads to hide type parameters from places where they are not necessary.